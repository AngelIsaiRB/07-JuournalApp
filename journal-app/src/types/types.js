

export const types={
    login: "[Auth] login",
    logout: "[Auth] logout",

    uiSetError: "[ui] set error",
    uiRemoveError: "[ui] remover error",

    uiStartLoading: '[UI] Start loading',
    uiFinishLoading: '[UI] Finish loading',

    notesAddNew: "[Notes] New note",
    notesActive: "[Notes] Set active note",
    notesLoad: "[Notes] Load note",
    notesUpdate: "[Notes] Update note",
    notesFileUrl: "[Notes] Update image url",
    notesDelete: "[Notes] Delete note",
    notesLogoutCleaning: "[Notes] Logout Cleaning",

}