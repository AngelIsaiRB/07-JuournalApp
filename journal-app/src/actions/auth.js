import { types } from "../types/types"
import {firebase, googleAuthProvider} from "../firebase/firebaseConfig"

export const startLoginEmailPassword = ()=>{
    return (dispatch) =>{
        setTimeout(() =>{
            dispatch(login(1234,"isai"));
        },3500)
    }
}

export const startGoogleLogin=()=>{
    return (dispatch)=>{
        firebase.auth().signInWithPopup(googleAuthProvider)
        .then(({user}) =>{
            dispatch(login(user.uid, user.displayName));
        });
    }
}

export const login =(uid, displayName)=>(
     {
        type: types.login,
        payload: {
            uid,
            displayName
        }
    })
