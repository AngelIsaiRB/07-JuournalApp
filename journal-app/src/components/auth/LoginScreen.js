import React from 'react'

import {useDispatch} from "react-redux";
import { Link } from 'react-router-dom'
import { login } from '../../actions/auth';
import { useForm } from '../../hooks/useForm'

export const LoginScreen = () => {

    const dispatch = useDispatch();

    const [formValues, hnaldeinputchange] = useForm({
        email: "email@emial.com"  ,
        password:"1q123412",
    });

    const {email, password}=formValues

    const handleLogin=(e)=>{
        e.preventDefault();
        dispatch(login(123456,"isai"));
    }


    return (
        <div>
            <h3 className="auth__title">Login</h3>
            <form onSubmit={handleLogin}>
                <input 
                    className="auth__input" 
                    value={email}   
                    onChange={hnaldeinputchange} 
                    autocomplete="off" 
                    type="text" 
                    placeholder="Email" 
                    name="email"></input>

                <input 
                    className="auth__input" 
                    value={password} 
                    onChange={hnaldeinputchange}  
                    type="password" 
                    placeholder="password" 
                    name="password"></input>

                <button className="btn btn-primary btn-block" type="submit" >Login</button>

                <hr/>
                <div className="auth__social-networks">
                    <p>login whit social network</p>
                    <div className="google-btn">
                        <div className="google-icon-wrapper">
                            <img className="google-icon" src="https://upload.wikimedia.org/wikipedia/commons/5/53/Google_%22G%22_Logo.svg" alt="google button" />
                        </div>
                        <p className="btn-text">
                        <b>Sign in with google</b>
                        </p>
                    </div>
                </div>
                <Link className="link" to="/auth/register">
                create new account
                </Link>
            </form>
        </div>
    )
}
