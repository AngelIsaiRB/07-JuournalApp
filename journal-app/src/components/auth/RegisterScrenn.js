import React from 'react';
import { useDispatch,useSelector } from 'react-redux';
import { Link } from 'react-router-dom';
import validator from "validator";
import { useForm } from '../../hooks/useForm';
import {  setError, removeError } from '../../actions/ui';

export const RegisterScrenn = () => {


    const dispatch = useDispatch();
    const {msgError} = useSelector(state=>state.ui);

    const  [formValues, handlenputChange ] = useForm({
        name: "isai",
        email : "email@emial.com",
        password : "password",
        password2 : "password",
    });

    const {name,email,password,password2} =formValues;

    const handleRegister = (e)=>{
        e.preventDefault();
        if(isFormValid()){

        }
    }

    const isFormValid = ()=>{
        
        if(name.trim().length === 0){ 
            // console.log("name is required");
            dispatch(setError("name is required"));
            return  false;
        }else if(!validator.isEmail(email)){
            // console.log("email is not valid");
            dispatch(setError("email is not valid"))
            return  false;
        }else if(password !== password2 || password.length <= 5){
            // console.log("Password should be at least 6 characters");     
            dispatch(setError("Password should be at least 6 characters"));
            return  false;
        }
        dispatch(removeError());
        return true;
        
    }

    return (
        <div>
            <h3 className="auth__title">Register</h3>
            <form onSubmit={handleRegister}>

               { 
               msgError &&
                (<div className="auth__alert-error">
                    {msgError}
                </div>)
                }

                <input value={name} onChange={handlenputChange} className="auth__input" autoComplete="off" type="text" placeholder="Name" name="name"></input>
                <input value={email} onChange={handlenputChange} className="auth__input" autoComplete="off" type="text" placeholder="Email" name="email"></input>
                <input value={password} onChange={handlenputChange} className="auth__input" type="password" placeholder="password" name="password"></input>
                <input value={password2} onChange={handlenputChange} className="auth__input" type="password" placeholder=" confirm password" name="password2"></input>

                <button className="btn btn-primary btn-block mb-5" type="submit" >Register</button>

               
               
                <Link className="link " to="/auth/login">
                Already register?
                </Link>
            </form>
        </div>
    )
}
