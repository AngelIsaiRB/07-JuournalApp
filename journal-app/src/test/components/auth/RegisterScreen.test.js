import configureStore from 'redux-mock-store'; 
import thunk from "redux-thunk";
import React from 'react';
import { shallow, mount } from 'enzyme'
import { RegisterScrenn } from '../../../components/auth/RegisterScrenn';
import { Provider } from 'react-redux';
import { MemoryRouter } from 'react-router-dom';
import { types } from '../../../types/types';

const middlewares = [thunk];
const mockStore = configureStore(middlewares);
const initState = {
    auth:{},
    ui:{
        loading: false,
        msgError: null,
    }
}
let store = mockStore(initState);


const wrapper = mount(
    <Provider store={store}>
        <MemoryRouter>
            <RegisterScrenn/>
        </MemoryRouter>
    </Provider>
        )

describe('pruebas en <RegisterScreen/>', () => {


    test('should  de hacer snapshot ', () => {
        expect(wrapper).toMatchSnapshot();
    });

    test('should de hacer el dispatch de la accion respectiva', () => {
        const emailField = wrapper.find("input[name='email']");
        emailField.simulate("change",{
            target:{
                value: '',
                name: 'email'
            }
        });
        wrapper.find("form").simulate("submit",{
            preventDefault(){}
        });
        const actions = store.getActions();
        expect(actions[0]).toEqual({
            type: types.uiSetError,
            payload: "email is not valid"
        })
        
    });

    test('should de mostrar la caja de alerta con el error', () => {
        const initState = {
            auth:{},
            ui:{
                loading: false,
                msgError: "email no es correcto",
            }
        }
        const store = mockStore(initState);
        
        
        const wrapper = mount(
            <Provider store={store}>
                <MemoryRouter>
                    <RegisterScrenn/>
                </MemoryRouter>
            </Provider>
        );
        expect(wrapper.find(".auth__alert-error").exists()).toBe(true);        
        expect(wrapper.find(".auth__alert-error").text().trim()).toBe(initState.ui.msgError);

    })
    
    

})