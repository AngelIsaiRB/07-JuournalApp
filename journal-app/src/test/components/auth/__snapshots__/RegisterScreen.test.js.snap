// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pruebas en <RegisterScreen/> should  de hacer snapshot  1`] = `
<div>
  <h3
    className="auth__title"
  >
    Register
  </h3>
  <form
    className="animate__animated animate__fadeIn animate__faster"
    onSubmit={[Function]}
  >
    <input
      autoComplete="off"
      className="auth__input"
      name="name"
      onChange={[Function]}
      placeholder="Name"
      type="text"
      value="isai"
    />
    <input
      autoComplete="off"
      className="auth__input"
      name="email"
      onChange={[Function]}
      placeholder="Email"
      type="text"
      value="email@emial.com"
    />
    <input
      className="auth__input"
      name="password"
      onChange={[Function]}
      placeholder="password"
      type="password"
      value="password"
    />
    <input
      className="auth__input"
      name="password2"
      onChange={[Function]}
      placeholder=" confirm password"
      type="password"
      value="password"
    />
    <button
      className="btn btn-primary btn-block mb-5"
      type="submit"
    >
      Register
    </button>
    <a
      className="link "
      href="/auth/login"
      onClick={[Function]}
    >
      Already register?
    </a>
  </form>
</div>
`;
